### PR TITLE
Orchagent send heartbeat during warm-reboot to prevent Orchagent stuck alert.

### PR DESCRIPTION
**Pull request by liuh-80**
Orchangent send heartbeat during warm-reboot to prevent Orchagent stuck alert.

#### Why I did it
Orchangent will freese during warm-reboot, then supervisor-proc-exit-listener will generate false alert during warm reboot:
https://github.com/sonic-net/sonic-buildimage/issues/16686

##### Work item tracking
- Microsoft ADO: 25295846

#### How I did it
Send heartbeat during warm-reboot freeze.

#### How to verify it
Pass all UT.
Manually verify issue fixed by check syslog.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [] 202205
- [] 202211
- [] 202305

#### Tested branch (Please provide the tested image version)

- [x]  Will update later

#### Description for the changelog
Orchangent send heartbeat during warm-reboot to prevent Orchagent stuck alert.

### CODE REVIEW

File: orchagent/orchdaemon.cpp [hunk 1]
```diff
@@ -863,7 +863,13 @@ void OrchDaemon::start()
                     flush();
 
                     SWSS_LOG_WARN("Orchagent is frozen for warm restart!");
-                    sleep(UINT_MAX);
+                    while (true)
+                    {
+                        // Send heartbeat message to prevent Orchagen stuck alert.
```
Comment (vaibhavhd) [inline orchagent/orchdaemon.cpp:868]: Typo in Orchagen

**Comment by vaibhavhd** [comment]
I think this qualifies for a new UT, to check of heartbeats are still sent after Orchagent pause.

File: orchagent/orchdaemon.cpp (hunk 1 above)
Comment (liuh-80) [reply orchagent/orchdaemon.cpp]: Thanks, Fixed.

**Comment by liuh-80** [comment]
> I think this qualifies for a new UT, to check of heartbeats are still sent after Orchagent pause.

I will check if I can add UT in this repo or sonic-mgmt repo.

**Comment by liuh-80** [comment]
Trigger build again

File: orchagent/orchdaemon.cpp [hunk 2]
```diff
@@ -864,7 +864,13 @@ void OrchDaemon::start()
                     flush();
 
                     SWSS_LOG_WARN("Orchagent is frozen for warm restart!");
-                    sleep(UINT_MAX);
+                    while (true)
+                    {
+                        // Send heartbeat message to prevent Orchagent stuck alert.
```
Comment (vaibhavhd) [inline orchagent/orchdaemon.cpp:869]: Just curious - at some point in time in the shutdown process, swss will be down. At that time will there still be OA stuck messages?

Comment (liuh-80) [reply orchagent/orchdaemon.cpp:869]: This message will send to systemd, then systemd will forward to process-exit-listener. 

process-exit-listener will monitor both heartbeat message and orchanget process exit message.

If the orchange is down, process-exit-listener will receive the process exit signal, then send process not running warning. in that case there will be no OA stuck message.

**Review by qiluo-msft — DISMISSED** [review]

**Comment by dgsudharsan** [comment]
@liuh-80 Can you please address the pipeline failures?

**Comment by liuh-80** [comment]
> @liuh-80 Can you please address the pipeline failures?

Sure, I'm still working on this PR.

**Comment by liuh-80** [comment]
The UT failed because supervisord not update process stdout to syslog immediately, it always delays few minutes, so test cause in this repo is very difficult to check the heartbeat signal.
Still trying to findout a solution to check heartbeat signal after freeze.

**Comment by prsunny** [comment]
@qiluo-msft , @vaibhavhd to signoff

**Comment by dgsudharsan** [comment]
@liuh-80  This PR conflicts when I cherry-pick to 202305 branch. Can you please create a new PR for fixing it in 202305?

**Comment by dgsudharsan** [comment]
@prsunny @qiluo-msft @vaibhavhd Can you please sign off this PR?

**Review by dgsudharsan — APPROVED** [review]

**Comment by prsunny** [comment]
> > I think this qualifies for a new UT, to check of heartbeats are still sent after Orchagent pause.
> 
> Fixed, add heartbeat message check to warm reboot UT.

@liuh-80 , i dont see the UT changes. Can you check?

**Comment by liuh-80** [comment]
> > > I think this qualifies for a new UT, to check of heartbeats are still sent after Orchagent pause.
> > 
> > 
> > Fixed, add heartbeat message check to warm reboot UT.
> 
> @liuh-80 , i dont see the UT changes. Can you check?

@prsunny , I try to add UT but found it's difficult to create a UT in this repo, because heartbeat message is send to systemd and sytemd have few minutes delay to write syslog.

So, my plan is to create test case in sonic-mgmt repo.

**Comment by liuh-80** [comment]
UT created in sonic-mgmt repo: https://github.com/sonic-net/sonic-mgmt/pull/10676
Will public UT for review after sonic-swss submodule update.

**Comment by StormLiangMS** [comment]
@liuh-80  could you help to update the test result with 202305?

**Comment by liuh-80** [comment]
> @liuh-80 could you help to update the test result with 202305?

Description updated, tested on SONiC.202305-17081.401641-ec2aed854

**Comment by StormLiangMS** [comment]
@liuh-80  cherry pick conflict, could you file separate PR to 202305?

**Comment by liuh-80** [comment]
> @liuh-80 cherry pick conflict, could you file separate PR to 202305?

202305 branch PR created: https://github.com/sonic-net/sonic-swss/pull/2956

**Pull request merged**